app/modules/inventory/components/shelf_dot.svelte

```diff
@@ -0,0 +1,57 @@
+<script>
+  import { isOpenedOutside } from '#lib/utils'
+  import { loadShelfLink } from '#shelves/components/lib/shelves'
+  import { getColorHexCodeFromModelId } from '#lib/images'
+  import screen_ from '#lib/screen'
+
+  export let shelf
+
+  const { name, _id } = shelf
+
+  const colorHexCode = shelf.color || `#${getColorHexCodeFromModelId(_id)}`
+
+  function onShelfDotClick (e, shelf) {
+    if (isOpenedOutside(e)) return
+    loadShelfLink(shelf)
+    e.preventDefault()
+  }
+</script>
+<li>
+  {#if screen_.isSmall(600)}
+    <div
+      class="shelf-dot"
+      title={name}
+      style:background-color="{colorHexCode}"
+    >
+    </div>
+  {:else}
+    <a
+      href="/shelves/{_id}"
+      on:click={e => onShelfDotClick(e, shelf)}
```

Comment:
(goes with the comment above)
```suggestion
      on:click={onShelfDotClick}
```